### DISCUSSION TITLE
tree_category

### DISCUSSION THREAD

**Discussion by tvideo** [totumonline/totum-ru-issues-and-discussions#157] [Как это сделать?] [answered]
Здравствуйте. Может есть более подробная информация, как использовать "tree_category", так как нужно иметь возможность выбирать строку, которая относится к папке.
1) В частности не понятно, где должно быть добавлено поле "tree_category", в главную таблицу-дерево, или в таблицу, из которой мы вызываем главное tree?
2) И если можно, поясните фразу "Для того, что бы они имели значения, необходимо создать поле с name = tree_category....В поле кодом или вручную должно быть выбрано значение соответсвующее id или bfield папки, которой соответсвует строка." Не могу понять какого типа должно быть поле "tree_category" и какое именно значение оно должно брать.
Спасибо

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Это непонятная фигня, тк связана она вот с чем:

У вас `tree-view` таблица с типом `tree` `«по другой таблице»`. В этом случае у вас строки с названиями категорий (папок) не являются строками таблицы — они такие виртуальные.

А бывают случаи, когда нам нужно, что бы категории тоже выглядели как строки.

Для этого в таблице в которой у вас поле `tree` делается поле `tree_category` и в нем указывается к какой категории относится строка.

![365_538_file](https://user-images.githubusercontent.com/55755565/173819762-56f73a4f-88e0-429d-be47-41a5494b01d5.png)

**Reply by ttm-alexeypolunin** [reply]
Все-равно немног мутно, будет видос чуть позже.

**Reply by DaveMaks** [reply]
> Все-равно немног мутно, будет видос чуть позже.

Очень ждем с нетерпением, второй день "курю" мануал, не могу собрать дерево-таблицу  при этом чтобы заголовки находились у папок. [https://totum.totum.online/fls/365_378_file.png](https://totum.totum.online/fls/365_378_file.png) 
Задача проста, сделать дерево структурных подразделений, и сотрудников, чтобы их потом можно было выбирать из дерева в других таблицах через "селект-дерево"

**Reply by ttm-alexeypolunin** [reply]
Это tree-view с типом «По другой таблице»: https://ru.docs.totum.online/treeview-based-on-select-tree

tree_category крайне редкий случай, когда нужно сделать вид «По другой таблице» похожим на «По текущей таблице»

К сожалению сейчас нет времени записывать видео, но обещаю вернуться к этому в марте.

**Comment by tvideo** [comment]
Может подскажите тогда, в какую сторону думать. Задача следующая:
Есть простая таблица с префильтром. В строчной части каждая относится к какому-нибудь разделу.
![image](https://user-images.githubusercontent.com/32614108/174645082-2840b872-33c5-42e7-ad3b-eefa281b3813.png)

глубина вложенности видна на скрине. Сейчас для поиска мне нужно выбрать конкретную вложенную папку.
Как сделать что бы в поиске в префильтре можно было выбрать "Видеонаблюдение" (то есть папка) и строчная часть отобразила все вложенные значения?

Может создать доп поле после tree, допустим "Родитель", и в нем указать родительскую папку, но как тогда организовать поиск?
Спасибо

**Reply by ttm-alexeypolunin** [reply]
Это делается через скрытое поле префильтра. Те у вас должно быть одно поле в котором вы чтот выбираете — оно не привязывается ни к чему.

Второе поле в префильтре — скрытое — оно привязывается. И в скрытом поле вы вычисляете все возможные значения, которые должны попасть в выборку.

Дочерние по дереву берет функция `selectTreeChildren`
